static/js/hotkey.js

```diff
@@ -308,6 +308,10 @@ export function process_escape_key(e) {
                 return true;
             }
 
+            if ($("#compose_banners").length) {
+                $("#compose_banners").empty();
+            }
```

Comment:
What's the thinking behind adding this here? We don't have a similar check for the preexisting banners system.

Might make sense to put this in a separate commit with a commit message explaining the change; and also it's probably worth having a comment here for posterity with some color for why we want to clear compose banners.

Though I think it's probably a bug to have this here and `compose_actions.cancel` or one of its subroutines should be responsible?

Reply:
We do a similar check for the preexisting banners system right above this.

```
            if ($("#compose-send-status").css("display") !== "none") {
                $("#compose-send-status").hide();
                return true;
            }
```

Reply:
That other version returns `true` -- which means it considers this the complete processing of the `Esc` keyboard shortcut (i.e. it'll just hide the banner, and not also close the compose box). Shouldn't this one do that? 

(And I guess we haven't ported 100% of things, so we don't get to delete the old one).

Reply:
Yup, the old one gets deleted in the upcoming Upload Banner PR, once we've stopped using `$("#compose-send-status")` completely :)

And yes good catch, I'll add the `return true`

Reply:
The `return true` got placed one commit after the code introducing this check, which seems like a rebase mistake. I've fixed that in my local copy but FYI, the best way to catch this sort of thing is to just look at the commits in `gitk` or a similar viewer after doing the rebase to check your work.